From 81d04ae4b8330c0788bf4c04c80367b527881079 Mon Sep 17 00:00:00 2001
From: David Faure <faure@kde.org>
Date: Fri, 4 Apr 2014 17:21:55 +0200
Subject: [PATCH 08/14] Add DesktopNames key to kdm sessions files.

This will be used by kdm (and other session managers) to set
XDG_CURRENT_DESKTOP when logging into such a desktop.

There is no written standard for session files but we agreed to this naming
and dm functinality at the freedesktop summit.
---
 kdm/kfrontend/sessions/gnome.desktop                 | 1 +
 kdm/kfrontend/sessions/kde-plasma-safe.desktop.cmake | 1 +
 kdm/kfrontend/sessions/kde-plasma.desktop.cmake      | 1 +
 kdm/kfrontend/sessions/lxde.desktop                  | 1 +
 kdm/kfrontend/sessions/xfce.desktop                  | 1 +
 kdm/kfrontend/sessions/xfce4.desktop                 | 1 +
 6 files changed, 6 insertions(+)

diff --git a/kdm/kfrontend/sessions/gnome.desktop b/kdm/kfrontend/sessions/gnome.desktop
index f4263dd..8746f80 100644
--- a/kdm/kfrontend/sessions/gnome.desktop
+++ b/kdm/kfrontend/sessions/gnome.desktop
@@ -2,6 +2,7 @@
 Type=XSession
 Exec=gnome-session
 TryExec=gnome-session
+DesktopNames=GNOME
 Name=GNOME
 Name[af]=GNOME
 Name[ar]=جنوم
diff --git a/kdm/kfrontend/sessions/kde-plasma-safe.desktop.cmake b/kdm/kfrontend/sessions/kde-plasma-safe.desktop.cmake
index 722091e..40881fe 100644
--- a/kdm/kfrontend/sessions/kde-plasma-safe.desktop.cmake
+++ b/kdm/kfrontend/sessions/kde-plasma-safe.desktop.cmake
@@ -3,6 +3,7 @@ Encoding=UTF-8
 Type=XSession
 Exec=${BIN_INSTALL_DIR}/startkde --failsafe
 TryExec=${BIN_INSTALL_DIR}/startkde
+DesktopNames=KDE
 Name=KDE Plasma Workspace (failsafe session)
 Name[ar]=مساحة عمل بلازما كدي (الجلسة الآمنة)
 Name[bn]=কে.ডি.ই. প্লাসমা ওয়ার্কস্পেস (failsafe session)
diff --git a/kdm/kfrontend/sessions/kde-plasma.desktop.cmake b/kdm/kfrontend/sessions/kde-plasma.desktop.cmake
index 37413b6..f13659c 100644
--- a/kdm/kfrontend/sessions/kde-plasma.desktop.cmake
+++ b/kdm/kfrontend/sessions/kde-plasma.desktop.cmake
@@ -3,6 +3,7 @@ Encoding=UTF-8
 Type=XSession
 Exec=${BIN_INSTALL_DIR}/startkde
 TryExec=${BIN_INSTALL_DIR}/startkde
+DesktopNames=KDE
 Name=KDE Plasma Workspace
 Name[ar]=مساحة عمل بلازما كدي
 Name[bg]=Работно пространство Plasma на KDE
diff --git a/kdm/kfrontend/sessions/lxde.desktop b/kdm/kfrontend/sessions/lxde.desktop
index b794fb2..3ad88bd 100644
--- a/kdm/kfrontend/sessions/lxde.desktop
+++ b/kdm/kfrontend/sessions/lxde.desktop
@@ -2,6 +2,7 @@
 Type=XSession
 Exec=startlxde
 TryExec=startlxde
+DesktopNames=LXDE
 Name=LXDE
 Name[ar]=LXDE
 Name[ast]=LXDE
diff --git a/kdm/kfrontend/sessions/xfce.desktop b/kdm/kfrontend/sessions/xfce.desktop
index c336230..1a11002 100644
--- a/kdm/kfrontend/sessions/xfce.desktop
+++ b/kdm/kfrontend/sessions/xfce.desktop
@@ -2,6 +2,7 @@
 Type=XSession
 Exec=xfwm
 TryExec=xfwm
+DesktopNames=XFCE
 Name=XFce
 Name[af]=XFce
 Name[ar]=XFce
diff --git a/kdm/kfrontend/sessions/xfce4.desktop b/kdm/kfrontend/sessions/xfce4.desktop
index 75c88a1..d4d7905 100644
--- a/kdm/kfrontend/sessions/xfce4.desktop
+++ b/kdm/kfrontend/sessions/xfce4.desktop
@@ -2,6 +2,7 @@
 Type=XSession
 Exec=startxfce4
 TryExec=startxfce4
+DesktopNames=XFCE
 Name=Xfce 4
 Name[ar]=XFce 4
 Name[bg]=Xfce 4
-- 
1.9.0

